caddyhttp/staticfiles/fileserver.go

```diff
@@ -121,13 +125,35 @@ func (fs FileServer) serveFile(w http.ResponseWriter, r *http.Request, name stri
 		return http.StatusNotFound, nil
 	}
 
+	filename := d.Name()
+
+	for _, encoding := range staticEncodingPriority {
+		if strings.Contains(r.Header.Get("Accept-Encoding"), encoding) {
+			gf, err := fs.Root.Open(location + staticEncoding[encoding])
+			if err == nil {
+				defer gf.Close()
```

Comment:
I think this defer could potentially leak file descriptors. If a file takes a long time to download (network congestion or client leaves connection open or something), this handler may never terminate and the loop that we're in could open multiple files that never end up being used.